lucetc/src/compiler.rs

```diff
@@ -72,14 +72,15 @@ impl<'a> Compiler<'a> {
         }
 
         translate_module(wasm_binary, &mut module_info).map_err(|e| match e {
+            WasmError::User(_) => e.context(LucetcErrorKind::Input),
```

Comment:
@pchickey I saw you mentioned wanting this on the corresponding cranelift issue, were you thinking about doing anything in particular with this variant? I have it contextualized as `Input` because I imagine this would be "user provided invalid input/options"

Reply:
I don't think we need to do anything special for now. In the future we may customize error reporting for invalid user-provided input but the way you've done it here is sufficient.